### ISSUE TITLE
Airodump-ng 1.2rc4 crashing

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1676]
**Reported by misterx on 16 Feb 2016 23:14 UTC**

When capturing with a rtl8187 device (or b43), running ```airodump-ng wlan0mon``` crashes on Kali 2016.1 with the following error in dmesg:

```
[traps: airodump-ng[31447](15471.089355]) trap invalid opcode ip:55c2a71f5773 sp:7ffcb42cd520 error:0 in airodump-ng[55c2a71ea000+27000]
```

On the screen, it just starts (blank screen) then crashes and displays ```Illegal instruction```.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 16 Feb 2016 23:14 UTC**

That is really weird because it works just fine for me. I tested a rtl8187 and ath9k_htc. No issue whatsoever.

Looking at the message makes me think it might be due on how it was compile and the fact that your CPU doesn't support certain instructions. And that means we only need to enable those special instructions for aircrack-ng only (and nothing else).

Could you post ```aircrack-ng -u``` output and also ```cat /proc/cpuinfo``` (or ```lscpu```)?

**Comment by aircrack-ng** [comment]
**Comment by rainfyre on 16 Feb 2016 23:14 UTC**

root@kali:~# aircrack-ng -u
Vendor          = Intel
Model           = Intel(R) Core(TM)2 Duo CPU     T7250  @ 2.00GHz
Features        = MMX,SSE,SSE2,SSE3,SSSE3
Hyper-Threading = No
CPU frequency   = 2001 MHz (Max: 2001 MHz)
Illegal instruction
root@kali:~#

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 16 Feb 2016 23:14 UTC**

root@kali:~# cat /proc/cpuinfo
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 15
model name	: Intel(R) Core(TM)2 Duo CPU     T7250  @ 2.00GHz
stepping	: 13
microcode	: 0xa4
cpu MHz		: 2000.000
cache size	: 2048 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 2
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 10
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm lahf_lm ida dtherm tpr_shadow vnmi flexpriority
bugs		:
bogomips	: 3989.99
clflush size	: 64
cache_alignment	: 64
address sizes	: 36 bits physical, 48 bits virtual
power management:

processor	: 1
vendor_id	: GenuineIntel
cpu family	: 6
model		: 15
model name	: Intel(R) Core(TM)2 Duo CPU     T7250  @ 2.00GHz
stepping	: 13
microcode	: 0xa4
cpu MHz		: 1200.000
cache size	: 2048 KB
physical id	: 0
siblings	: 2
core id		: 1
cpu cores	: 2
apicid		: 1
initial apicid	: 1
fpu		: yes
fpu_exception	: yes
cpuid level	: 10
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm lahf_lm ida dtherm tpr_shadow vnmi flexpriority
bugs		:
bogomips	: 3989.99
clflush size	: 64
cache_alignment	: 64
address sizes	: 36 bits physical, 48 bits virtual
power management:

root@kali:~#

**Comment by aircrack-ng** [comment]
**Comment by misterx on 16 Feb 2016 23:14 UTC**

Yes, this is exactly what I was thinking.

We can close this when #1679 is implemented.

**Comment by aircrack-ng** [comment]
**Comment by Araino on 16 Feb 2016 23:14 UTC**

Hello. I have been having the same error and just wanted to post my results for some feedback/to help resolve this issue. 

```airmon-ng start wlan0```

```python
PHY	Interface	Driver		Chipset

phy0	wlan0		iwlwifi		Intel Corporation WiFi Link 5100

		(mac80211 monitor mode vif enabled for [on [phy0](phy0]wlan0)wlan0mon)
		(mac80211 station mode vif disabled for [wlan0mon```
```python
(blank)
Illegal instruction
```

```dmesg```
```python
[  525.139124](phy0]wlan0)
```

```airodump-ng) device wlan0mon entered promiscuous mode
[  525.301982] traps: airodump-ng[trap invalid opcode ip:565188495773 sp:7ffff0af4780 error:0 in airodump-ng[56518848a000+27000](2935])
```

```aircrack-ng -u```
```python
Vendor          = Intel
Model           = Intel(R) Core(TM)2 Duo CPU     T9550  @ 2.66GHz
Features        = MMX,SSE,SSE2,SSE3,SSSE3,SSE4.1
Hyper-Threading = No
CPU frequency   = 2133 MHz (Max: 2667 MHz)
Illegal instruction
```

```lscpu```
```python
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                2
On-line CPU(s) list:   0,1
Thread(s) per core:    1
Core(s) per socket:    2
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 23
Model name:            Intel(R) Core(TM)2 Duo CPU     T9550  @ 2.66GHz
Stepping:              10
CPU MHz:               800.000
CPU max MHz:           2667.0000
CPU min MHz:           800.0000
BogoMIPS:              5320.55
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              6144K
NUMA node0 CPU(s):     0,1
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl aperfmperf pni dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm sse4_1 xsave lahf_lm ida dtherm tpr_shadow vnmi flexpriority
```

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 16 Feb 2016 23:14 UTC**

Illegal instruction Im also getting this... Fuck it Ill stick with stable version until a new release is out these beta's are more hassle than there worth, Beta's are testing ground but when there unfunctional its not worth using them

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 16 Feb 2016 23:14 UTC**

Illegal instruction (core dumped)

I've tried aircrack-1.2-rc4 with broadcom,ralink I never had a problem with these cards in 1.2-rc3

**Comment by aircrack-ng** [comment]
**Comment by misterx on 16 Feb 2016 23:14 UTC**

Compile it yourself as a temporary workaround.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 16 Feb 2016 23:14 UTC**

Should be fixed by aircrack-ng/aircrack-ng@4472d7dec98133e6f0df336457f2caf1c73d9842.
